packages/web/src/react.tsx

```diff
@@ -25,14 +25,27 @@ import type { AnalyticsProps } from './types';
  * }
  * ```
  */
-function Analytics({
-  beforeSend,
-  debug = true,
-  mode = 'auto',
-}: AnalyticsProps): null {
+function Analytics(
+  props: AnalyticsProps & {
+    framework?: string;
+    path?: string | null;
+  }
+): null {
   useEffect(() => {
-    inject({ beforeSend, debug, mode });
-  }, [beforeSend, debug, mode]);
+    inject({
+      framework: props.framework || 'react',
+      ...(props.route !== undefined && { disableAutoTrack: true }),
+      ...props,
+    });
+  }, [props]);
+
+  useEffect(() => {
+    if (props.route) {
+      pageview({
+        route: props.route,
+      });
```

Comment:
**question(blocking):** do you think we should update `data-` attribute (or anything else) so we could pick the current route when sending custom events with `track()` function?

Reply:
that is already handled internally in the script. we store the current `route` as internal variable that will be passed on each custom event